docs/docs/using-the-template.md

```diff
@@ -0,0 +1,158 @@
+Writing objects: 100% (23/23), 6.47 KiB | 6.47 MiB/s, done.
+Total 23 (delta 0), reused 0 (delta 0), pack-reused 0
+To https://github.com/pjbull/testccds.git
+ * [new branch]      HEAD -> main
+branch 'main' set up to track 'origin/main'.
+✓ Pushed commits to https://github.com/pjbull/testccds.git
+```
+
+## Create a Python virutal environment
+
+We often use Python for our data science projects. We use a virtual environment to manage the packages we use in our project. This is a way to keep the packages we use in our project separate from the packages we use in other projects. This is especially important when we are working on multiple projects at the same time.
+
+Cookicutter Data Science supports a few options for virtual environment management, but no matter which you choose, you can create an environment with the following commands:
+
+```bash
+make create_environment
+```
+
+Once the environment is created, you'll want to make sure to activate it. You'll have to do this following the instructions for your specific environment manager. We recommend using a shell prompt that shows you which environment you are in, so you can easily tell if you are in the right environment, for example [starship](https://starship.rs/). You can also use the command `which python` to make sure that your shell is pointing to the version of Python associated with your virtual environment.
+
+Once you are sure that your environment is activated in your shell, you can install the packages you need for your project. You can do this with the following command:
+
+```bash
+make requirements
+```
+
+## Add your data
+
+There's no universal advice for how to manage your data, but here are some recommendations for starting points depending on where the data comes from:
+
+ - **Flat files (e.g., CSVs or spreadsheets) that are static** - Put these files into your `data/raw` folder and then run `make sync_data_up` to push the raw data to your cloud provider.
+ - **Flat files that change and are extracted from somewhere** - Add a Python script to your source module in `data/make_dataset.py` that downloads the data and puts it in the `data/raw` folder. Then you can use this to get the latest and push it up to your cloud host as it changes.
+ - **Databases you connect to with credentials** - Store your credentials in `.env`. We recommend adding a `db.py` file or similar to your `data` module that connects to the database and pulls data. If your queries generally fit into memory, you can just have functions in the `db.py` to load data that you use in analysis. If not, you'll want to add a script like above to download the data to the `data/raw` folder.
+
+## Check out a branch
+
+We'll talk about code review later, but it's a good practice to use feature branches and pull requests to keep your development organized. Now that you have source control configured, you can check out a branch to work in with:
+
+```
+git checkout -b initial-exploration
+```
+
+## Open a notebook
+
```

Comment:
We're mostly agnostic about notebooks but this section uses Jupyter, may be worthwhile to have something like

```suggestion

!!! note 

    The following assumes you're using a Jupyter notebook, but while the specific commands for another notebook tool may look a little bit different, the process guidance still applies.

```